### ISSUE TITLE
No cluster leadership acquired

### ISSUE THREAD

**Issue by aromm** [victorcoder/dkron#133]
Hi there,

I am very new to dkron and am currently in the process of setting up a cluster. By now I have been able to join my three nodes but the leader election process is not working.

My nodes all have the following dkron config file:
```
{
  "backend": "consul", 
  "backend_machine": "127.0.0.1:8500",
  "server": true,
  "log-level": "debug",
  "tags": {
    "role": "test",
    "datacenter": "east"
  },
  "keyspace": "cron",
  "encrypt": "{{ encrypt value }}",
  "http_addr": ":8080",
  "join": [
    "10.64.0.23",
    "10.64.0.26",
    "10.64.0.28"
  ]
}
```

I have determined that the three nodes do "find" each other and the dashboard displays all three nodes as well. However the dashboard nor `curl http://127.0.0.1:8500/v1/leader` show any leader.

I have been trying to figure out the problem and it turns out I can manually run bash scripts on all three clusters using the API and clicking "Run" on the dashboard/jobs page and they successfully run/execute (the script emails me and I get an email but the dashboard graph also says the scripts are getting errors." When I run a job I can even find they kv store by running `curl 127.0.0.1:8500/v1/kv/?recurse` and every execution and job shows up.

I read through some of the code and found that there should be a consul key created called cron/leader but it has not been created. Also I found that there should have been a new consul session created in `NewLock()` with no name but I could not find the consul session either. 

The dkron.log file shows on all 3 nodes
```
time="2016-05-31T17:10:00Z" level=info msg="agent: Cluster leadership lost" node=ip-10-64-0-28 
time="2016-05-31T17:10:10Z" level=info msg="agent: Running for election" node=ip-10-64-0-28 
time="2016-05-31T17:10:10Z" level=info msg="agent: Cluster leadership lost" node=ip-10-64-0-28 
time="2016-05-31T17:10:20Z" level=info msg="agent: Running for election" node=ip-10-64-0-28 
time="2016-05-31T17:10:20Z" level=info msg="agent: Cluster leadership lost" node=ip-10-64-0-28 
time="2016-05-31T17:10:31Z" level=info msg="agent: Running for election" node=ip-10-64-0-28 
time="2016-05-31T17:10:31Z" level=info msg="agent: Cluster leadership lost" node=ip-10-64-0-28 
```
with the occasional 
```
time="2016-05-27T21:33:19Z" level=fatal msg="Unexpected response code: 500" node=ip-10-64-0-28 
```

I am using consul v.0.6.3 protocol 3 and dkron v.0.7.1

I am very confused because it seems like dkron is working with consul well but I dont understand why the session still was not created even though kv stores for jobs are being created. Any help/advice would be appreciated. I am not entirely sure if I set this up properly but I could not find solutions anywhere else.

**Comment by aromm** [comment]
I found in the consul logs
` 2016/05/31 19:29:28 [ERR] http: Request PUT /v1/session/create, error: rpc error: Invalid Session TTL '7500000000', must be between [10s=24h0m0s] from=127.0.0.1:50608 `

which seems to be the problem with why the session is not being created. I am not sure how dkron that can be since`defaultLeaderTTL   = 20 * time.Second`

**Comment by alekitto** [comment]
Same error here. Running consul v0.6.4 and dkron v0.7.1
Leader election works downgrading dkron to v0.7.0

**Comment by victorcoder** [comment]
@aromm @alekitto That's really weird, the only recent TTL adjust was in https://github.com/victorcoder/dkron/pull/120 and it increased from 15s to 20s

This should have something to do with the consul version, I'm using an outdated docker image.

I'll check this

**Comment by aromm** [comment]
@victorcoder I am running consul along dkron in the same 3 nodes however, I think I figured out the issue (haven't fixed it yet so cannot be sure). 

Firstly, I looked into #120 and the change you made was 14 days ago while v0.7.1 was released 29 days ago so for me the TTL is still 15s. 

Secondly, [https://github.com/docker/swarm/issues/2153](url) mentions that the `7500000000` value is in nanoseconds which is equal to 7.5s. This 7.5s is the value of 15s dkron passes in but then Consul takes the TTL value and divides it by two hence 15/2 = 7.5 s = 7500000000 ns.

So #120 should fix everything, I just need to the fix in my running version I guess.
